src/server/snapshot.cc

```diff
@@ -243,6 +243,7 @@ void SliceSnapshot::SwitchIncrementalFb(Context* cntx, LSN lsn) {
 }
 
 bool SliceSnapshot::BucketSaveCb(PrimeIterator it) {
+  ConditionGuard guard(&bucket_ser_);
```

Comment:
why here and in OnDbChange and not inside SerializeBucket ?

Reply:
You have asked me the exact same question in June :rofl:

The answer is, that if you look on `BucketSaveCb` it calls `FlushChangeToEarlierCallbacks` which might preempt and when the fiber resumes the condition `the check` above might not be true. The code within `BucketSaveCb` should be atomic /considered as a critical section

Reply:
do you mean the check below lambda for version?
you can check also inside BucketSaveCb the snapshot version and if its bigger than dont serialize the bucket